### ISSUE TITLE
Is there an option to supply a bed file with particular regions for phasing ?

### ISSUE THREAD

**Issue by ilivyatan** [twolinin/longphase#31]
Is there an option to specify regions for phasing rather than the entire genome?
Or do I have to preprocess the BAM to specific regions?

**Comment by twolinin** [comment]
You can preprocess the BAM to specific regions.

**Comment by ythuang0522** [comment]
No. We thought the speed is fast enough for chromosome phasing without the need for region sampling. You can either extract the alignments from the bam or variants from the vcf in your interested region by samtools/tabix.

**Comment by sloth-eat-pudding** [comment]
We have released version 1.7.
Haplotag now includes the `--region` feature.

input format:

- chrom (consider entire chromosome)
- chrom:start (consider region from this start to end of chromosome)
- chrom:start-end

**Comment by ilivyatan** [comment]
I want to expand the discussion on "focused regions".
From reading the paper, it seems that the algorithm is implemented as a dynamic program, when given a SNP, the "next" SNP is connected up, so a new SNP can either join or not join a growing phase block. Rather than examining the relationship between each SNP to every other one (in a given vicinity).
I'm wondering what would happen if the traversal over the genome were in the reverse order?? I can test this, but thought that it may have already come up. Do the phase blocks stay exactly the same?
What I'm trying to get at is the reliability of the algorithm on the context of the "anchor SNP". What happens if I shift that context slightly - start traversal from a different direction, or limit the SNPs to a given region.
If I've misunderstood the algorithm, then never mind. 
It would help to see a quantification of how deterministic the phasing is on the same SNP data.
